cachecontrol/caches/file_cache.py

```diff
@@ -79,13 +100,21 @@ def _fn(self, name):
         parts = list(hashed[:5]) + [hashed]
         return os.path.join(self.directory, *parts)
 
+    def _lockname(self, name):
+        if isinstance(self.lock_class, PYLockFileBase):
+            return self.name
+        else:
+            return self.name + ".lock"
+
     def get(self, key):
         name = self._fn(key)
-        if not os.path.exists(name):
-            return None
 
-        with open(name, 'rb') as fh:
-            return fh.read()
+        with self.lock_class(self._lockname(name)):
+            if not os.path.exists(name):
```

Comment:
Should this be refactored as a try/except block, instead of testing for path existence?